### PR TITLE
[RFC] Task Middleware Demo

### PR DESCRIPTION
**Pull request by gcoonrod**
This PR demonstrates a very basic implementation of middleware for tasks. Its very nearly a direct code copy from the action middleware implementation. 

I'm looking for comments and suggestions before I go any further.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -37,6 +37,7 @@
     "i18n": "^0.8.1",
     "ioredis": "^1.15.1",
     "is-running": "^2.0.0",
+    "lodash": "^4.13.1",
```
Comment (evantahler) [inline package.json]: never!

File: initializers/taskMiddleware.js [hunk 2]
```diff
@@ -0,0 +1,71 @@
+'use strict';
```
Comment (evantahler) [inline initializers/taskMiddleware.js]: Not sure what this file does?

**Comment by evantahler** [comment]
good start!

File: package.json (hunk 1 above)
Comment (l0oky) [inline package.json]: What's wrong with `lodash`?

File: initializers/demoMiddleware.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    api.demoMiddleware = {
+      name: "demoMiddleware",
+      priority: 999,
+      preProcessor: function(data, next){
+        api.log("In Preprocessor!", "info");
+        next(null, true);
+      },
+      postProcessor: function(data, next){
+        api.log("In Postprocessor!", "info");
+        api.log("Data keys: ", "info", _.keys(data));
```
Comment (S3bb1) [inline initializers/demoMiddleware.js]: you only use lodash for keys ? 

then you can remove the dependency and use Object.keys(data)

Comment (gcoonrod) [inline initializers/demoMiddleware.js]: This initializer is only there as a demo. It'll be removed completely.

File: package.json (hunk 1 above)
Comment (gcoonrod) [inline package.json]: This dep will be removed before the final PR is ready.

File: initializers/taskMiddleware.js (hunk 2 above)
Comment (gcoonrod) [inline initializers/taskMiddleware.js]: This is the resque plugin that handles the middleware processing.

**Comment by evantahler** [comment]
bump! 
Any progress here?

**Comment by evantahler** [comment]
also, please change this PR to merge into the `v14.0.0` branch, not master

**Comment by evantahler** [comment]
I'm going to cut v14 now... which means this isn't going to make it.   We can spend more time on this, and it will make it into v15 then

**Comment by gcoonrod** [comment]
Closing request after talking with @evantahler. I will cut a new PR from the current version in the near future. :pray:

**Pull request closed without merging**